### PR TITLE
Commit existing othello code

### PR DESCRIPTION
**Pull request by neuromantic99**

### CODE REVIEW

File: game_mechanics.py [hunk 1]
```diff
@@ -1,71 +1,449 @@
...
-    pass
+HERE = Path(__file__).parent.resolve()
 
 
-def load_pkl(team_name: str):
-    """Load a user PKL."""
-    pass
+def load_network(team_name: str) -> nn.Module:
+    net_path = os.path.join(HERE, f"{team_name}_network.pt")
+    assert os.path.exists(
```
Comment (Henry-Pulver) [inline game_mechanics.py]: Something something `pathlib` something

File: game_mechanics.py [hunk 2]
```diff
@@ -1,71 +1,449 @@
-from typing import Any, Callable, Dict, Tuple
+import copy
+import math
+import os
```
Comment (Henry-Pulver) [inline game_mechanics.py]: GRRRRR

File: game_mechanics.py [hunk 3]
```diff
@@ -1,71 +1,449 @@
...
-        """
+def make_move(board: np.ndarray, move: Tuple[int, int]) -> np.ndarray:
+    """Returns board after move has been made by player 1."""
+    return _make_move(board, move, 1)
+
+
+def choose_move_randomly(
+    board: np.ndarray,
+) -> Optional[Tuple[int, int]]:
+    """Returns a random legal move on the current board (always plays as player 1."""
```
Comment (Henry-Pulver) [inline game_mechanics.py]: Missing `)`

File: game_mechanics.py [hunk 4]
```diff
@@ -1,71 +1,449 @@
...
+    else:
+        raise ValueError(f"Move {move} is not a valid move!")
+    return board_after_move
+
+
+def is_legal_move(board: np.ndarray, move: Tuple[int, int], current_player: int) -> bool:
+    """Method: is_legal_move
+    Parameters: self, move (tuple)
+    Returns: boolean (True if move is legal, False otherwise)
+    Does: Checks whether the player's move is legal.
```
Comment (Henry-Pulver) [inline game_mechanics.py]: Strange format & out of date

Aren't we using Google docstrings?

File: game_mechanics.py [hunk 5]
```diff
@@ -1,71 +1,449 @@
...
+                return True
+    return False
+
+
+def is_valid_coord(board_dim: int, row: int, col: int) -> bool:
+    """Method: is_valid_coord
+    Parameters: self, row (integer), col (integer)
+    Returns: boolean (True if row and col is valid, False otherwise)
+    Does: Checks whether the given coordinate (row, col) is valid.
+            A valid coordinate must be in the range of the board.
```
Comment (Henry-Pulver) [inline game_mechanics.py]: Again, docstring out of date and weird format

File: game_mechanics.py [hunk 6]
```diff
@@ -1,71 +1,449 @@
...
+    Parameters: self, move (tuple), direction (tuple)
+    Returns: boolean
+                (True if there is any tile to flip, False otherwise)
+    Does: Checks whether the player has any adversary's tile to flip
+            with the move they make.
+
+            About input: move is the (row, col) coordinate of where the
+            player makes a move; direction is the direction in which the
+            adversary's tile is to be flipped (direction is any tuple
+            defined in MOVE_DIRS) -> None.
```
Comment (Henry-Pulver) [inline game_mechanics.py]: 3 for 3

File: game_mechanics.py [hunk 7]
```diff
@@ -1,71 +1,449 @@
...
+
+    return i > 1
+
+
+def flip_tiles(board: np.ndarray, move: Tuple[int, int], current_player: int) -> np.ndarray:
+    """Flips the adversary's tiles for current move and updates the running tile count for each
+    player.
+
+    Arg:
+        move (Tuple[int, int]): The move just made to
```
Comment (Henry-Pulver) [inline game_mechanics.py]: I'd cut types from the docstring in general. It's repeated information and therefore unnecessary. Also means this line doesn't need to wrap

File: game_mechanics.py [hunk 8]
```diff
@@ -1,71 +1,449 @@
...
+    return board
+
+
+def has_legal_move(board: np.ndarray, current_player: int) -> bool:
+    """Method: has_legal_move
+    Parameters: self
+    Returns: boolean
+                (True if the player has legal move, False otherwise)
+    Does: Checks whether the current player has any legal move
+            to make.
```
Comment (Henry-Pulver) [inline game_mechanics.py]: Another weird docstring

File: game_mechanics.py [hunk 9]
```diff
@@ -1,71 +1,449 @@
...
+    #             )
+
+    #             pygame.gfxdraw.filled_circle(
+    #                 self._screen,
+    #                 c * self.SQUARE_SIZE + self.SQUARE_SIZE // 2,
+    #                 r * self.SQUARE_SIZE + self.SQUARE_SIZE // 2,
+    #                 int(self.DISC_SIZE_RATIO * self.SQUARE_SIZE / 2),
+    #                 colour,
+    #             )
+    #     pygame.display.update()
```
Comment (Henry-Pulver) [inline game_mechanics.py:380]: TODO now or TODO later?

Side note: This VScode plugin may be useful for you to track TODO comments
https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.todo-tree

File: game_mechanics.py [hunk 10]
```diff
@@ -1,71 +1,449 @@
...
+    while not done:
+        action = your_choose_move(state)
+        state, reward, done, info = game.step(action, verbose)
+        # if render:
+        #     game.render()
+        total_return += reward
+        sleep(1 / game_speed_multiplier)
+
+    return total_return
+    # return info
```
Comment (Henry-Pulver) [inline game_mechanics.py]:
```suggestion

```

**Review by Henry-Pulver — COMMENTED** [review]
Nothing major - looks excellent!

Glad to see these getting a home of their own :smile:

File: game_mechanics.py (hunk 4 above)
Comment (neuromantic99) [reply game_mechanics.py]: yeah it's called the "Copy and pasted from the internet" formar

File: game_mechanics.py (hunk 9 above)
Comment (neuromantic99) [reply game_mechanics.py:380]: It's kinda important but no one complained at the lack of render option in last cohort

Comment (Henry-Pulver) [reply game_mechanics.py:380]: Add a ticket?

**Pull request merged**